### ISSUE TITLE
tlog: Vdisks should be deleted when no connection references it

### ISSUE THREAD

**Issue by robvanmieghem** [g8os/blockstor#221]
This is tricky as we also need to flush the unflushed blocks; 

And of course handle the case where we have missing blocks sequence;

**Comment by GlenDC** [comment]
> This is tricky as we also need to flush the unflushed blocks;

What do you mean with that? If there are unflushed blocks while not having any connection, surely that means that the tlog is in an invalid state...

> Vdisks should be deleted when no connection references it

And I guess you are just talking about the `*vdisk` instance in the `tlogserver` code, stored in the vdiskmanager?

**Comment by robvanmieghem** [comment]
@GlenDC Just copied what you typed in #167

**Comment by GlenDC** [comment]
> @GlenDC Just copied what you typed in #167

Yes I realized that and deleted my comment, haha. Damn, when reading that without any context that was really not clear at all to me, hahaha. Sorry for that.

**Comment by iwanbk** [comment]
previous discussion (for more context) https://github.com/g8os/blockstor/pull/168#issuecomment-301997206

**Comment by iwanbk** [comment]
> This is tricky as we also need to flush the unflushed blocks;

I found that this is not tricky anymore
